### ISSUE TITLE
Converting geant4 to FLUKA with reflections

### ISSUE THREAD

**Issue by stewartboogert** [g4edge/pyg4ometry#81]
@lnevay @FabiMetzger 

Currently not all geant4 solids can be converted with mxx, myy, mzz != 1 (where mij is the transformation matrix elements)

Only extruded solids are currently supported. Requires a complete set of tests and modification of the conversion code. What we have in the (for extruded only) conversion now is 

```
                  fbody = flukaRegistry.makeBody(
                      PLA,
                      "B" + name + format(ibody, "02"),
                      [
                          -nx * gdmlReflection[0],
                          -ny * gdmlReflection[1],
                          -nz * gdmlReflection[2],
                      ],
                      [
                          x0 * gdmlReflection[0],
                          y0 * gdmlReflection[1],
                          z0 * gdmlReflection[2],
                      ],
                      transform=transform,
                      flukaregistry=flukaRegistry,
                      comment=commentName,
                  )
```

Code like this will have to be replicated for all geant4 solids

**Comment by stewartboogert** [comment]
This problem is much more complex than first presented. The algorithm to extract reflections which a large hierarchy of  rotations/translations/scales(i.e reflections) is generally correct. There are some differences in qr decomposition (which might not be a stable solution).

**Comment by stewartboogert** [comment]
Upon further investigation this is much more complex. Generally extracting the reflections is not going to work. Particularly when there is compounded transformations. 

There are two types of geometry vectors which are vectors (E.g position of a plane) and pseudo vectors (E.g normal of a plane). Reviewing fluka bodies it should be possible to categorise vectors into the two classes and then transform the pseudo-vectors with the transformation determinant. 

This should preserve the sense of what is inside and outside of a region and allow reflections in composite transformations.

**Comment by stewartboogert** [comment]
Fixed by baking transforms in #83
